File: Tyche.Android/data/pool/src/main/java/com/felipearpa/data/pool/domain/PoolGamblerScoreRemoteDataSource.kt
package com.felipearpa.data.pool.domain

import com.felipearpa.tyche.core.paging.CursorPage
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

internal interface PoolGamblerScoreRemoteDataSource {
    @GET("/gamblers/{gamblerId}/pools")
    suspend fun getPoolGamblerScoresByGambler(
        @Path("gamblerId") gamblerId: String,
        @Query("next") next: String? = null,
        @Query("searchText") searchText: String? = null
    ): CursorPage<PoolGamblerScoreResponse>

    @GET("/pools/{poolId}/gamblers")
    suspend fun getPoolGamblerScoresByPool(
        @Path("poolId") poolId: String,
        @Query("next") next: String? = null,
        @Query("searchText") searchText: String? = null
    ): CursorPage<PoolGamblerScoreResponse>

    @GET("/pools/{poolId}/gamblers/{gamblerId}")
    suspend fun getPoolGamblerScore(
        @Path("poolId") poolId: String,
        @Path("gamblerId") gamblerId: String
    ): PoolGamblerScoreResponse
}

File: Tyche.iOS/DataPool/Sources/DataPool/Domain/PoolGamblerScoreRemoteDataSource.swift
import Core

protocol PoolGamblerScoreRemoteDataSource {
    func getPoolGamblerScoresByGambler(
        gamblerId: String,
        next: String?,
        searchText: String?
    ) async throws -> CursorPage<PoolGamblerScoreResponse>
    
    func getPoolGamblerScoresByPool(
        poolId: String,
        next: String?,
        searchText: String?
    ) async throws -> CursorPage<PoolGamblerScoreResponse>
    
    func getPoolGamblerScore(
        poolId: String,
        gamblerId: String
    ) async throws -> PoolGamblerScoreResponse
}
